crates/cli/src/updater.rs

```diff
@@ -444,17 +447,14 @@ impl Updater {
     async fn download_artifact(&self, app: SupportedApp, artifact_url: String) -> Result<PathBuf> {
         let target_path = self.bin_path.join(format!("{}-temp", app.get_bin_name()));
 
-        // Download via curl
-        let cmd = AsyncCommand::new("curl")
-            .arg("-L")
-            .arg("-o")
-            .arg(&target_path)
-            .arg(&artifact_url)
-            .output()
-            .await?;
-
-        if !cmd.status.success() {
-            bail!("Failed to download artifact: {:?}", cmd);
+        match reqwest::get(&artifact_url).await {
+            Ok(response) => {
+                let contents = response.bytes().await?.to_vec();
+                async_fs::write(&target_path, contents).await?;
```

Comment:
Is there a way to stream directly into the file, so we don't have to load it all into memory?

Reply:
We can do that! Streaming support is an extra feature in reqwest; are you okay with this expanding the dependency tree a little?

Reply:
That's fine, we're not very disciplined about binary size.

Reply:
All right, pushed a change that streams the response and writes chunks to disk.

Reply:
Thanks, looks good!